tests/install_upgrade_operators/network_policy/test_network_policy_components.py

```diff
@@ -0,0 +1,307 @@
+    pod_names = [pod.name for pod in pods if getattr(pod.instance.status, "phase", "") == "Running"]
+    assert len(pod_names) > 0, (
+        f"No running pods found matching NetworkPolicy {network_policy_by_name.name} podSelector "
+        f"in namespace {hco_namespace.name}"
+    )
+    pod_names.sort()
+    return pod_names
+
+
+@pytest.fixture
+def ingress_connectivity_baseline(
+    deployed_client_pod, service_ip, service_port, hco_namespace, admin_client, network_policy_match_labels
+):
+    """
+    Test preparation for ingress tests: Apply allow-all NetworkPolicy, verify traffic works, then remove it.
+    This ensures that without restrictive policies, the connection should succeed.
+    """
+    with AllowAllNetworkPolicy(
+        name="test-allow-all-traffic",
+        namespace=hco_namespace.name,
+        client=admin_client,
+        match_labels=network_policy_match_labels,
+    ):
+        try:
+            deployed_client_pod.execute(
+                command=shlex.split(f"curl -sS -k --connect-timeout 5 https://{service_ip}:{service_port}")
+            )
+        except Exception as e:
+            pytest.fail(
+                f"Test preparation FAILED: With allow-all NetworkPolicy, connection should succeed but got error: {e}"
+            )
+    yield {
+        "deployed_client_pod": deployed_client_pod,
+        "service_ip": service_ip,
+        "service_port": service_port,
+        "hco_namespace": hco_namespace,
+    }
+
+
+@pytest.fixture
+def egress_connectivity_baseline(
+    pods_by_network_policy, hco_namespace, admin_client, network_policy_match_labels, deployed_server_service
+):
+    """
+    Test preparation for egress tests: Apply allow-all NetworkPolicy, verify traffic works, then remove it.
+    This ensures that without restrictive policies, the connection should succeed.
+    """
+    with AllowAllNetworkPolicy(
+        name="test-allow-all-traffic",
+        namespace=hco_namespace.name,
+        client=admin_client,
+        match_labels=network_policy_match_labels,
+    ):
+        pod_name = pods_by_network_policy[0]
+        service_ip = (
+            deployed_server_service.instance.spec.get("clusterIPs")
+            or [deployed_server_service.instance.spec.get("clusterIP")]
+        )[0]
+        if ":" in service_ip:
+            pytest.skip(f"This test does not support sending traffic to IPv6: {service_ip}")
+
+        # Get Pod object to use execute() method
+        test_pod = Pod(client=admin_client, namespace=hco_namespace.name, name=pod_name)
+        try:
+            test_pod.execute(
+                command=shlex.split(f"curl -sS --connect-timeout 5 http://{service_ip}:{TEST_SERVER_PORT}")
+            )
+        except Exception as e:
+            pytest.fail(
+                f"Test preparation FAILED (egress): With allow-all NetworkPolicy, "
+                f"connection should succeed but got error: {e}"
+            )
```

Comment:
it looks very similar to previous test, how about to create a function and reuse?